App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListUserThreads/ListUserThreadsRequest.swift

```diff
@@ -0,0 +1,49 @@
+// system
+import Foundation
+import HTTPTypes
+
+// local
+import Onion
+
+struct GetListUserThreads: Request {
+    typealias Output = ListUserThreads
+    
+    var path: String {
+        return preparePathWithComponents()
+    }
+    
+    var headerFields: HTTPFields {
+        [
+            HTTPField.Name.authorization : "Bearer \(token)",
+            HTTPField.Name.contentType : "application/vnd.allegro.public.v1+json"
+        ]
+    }
+    
+    func preparePathWithComponents() -> String {
+        var components = URLComponents()
+        let defaultPath = "/messaging/threads"
+        
+        components.path = defaultPath
+        components.queryItems = []
+        
+        let queryItems: [(name: String, value: String?)] = [
+            ("limit", String(limit)),
+            ("offset", String(offset)),
+        ]
+        
+        components.queryItems?.append(contentsOf: queryItems.compactMap { name, value in
+            value.map { URLQueryItem(name: name, value: $0) }
+        })
+        
+        if let componentsURL = components.url?.absoluteString {
+            return componentsURL
+        } else {
+            return defaultPath
+        }
+    }
+    
+    let token: String
+    
+    var offset: Int = 0
```

Comment:
Checkout `PaginatedRequest` on the main branch :)

Reply:
Fixed :)